src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaFactoryTests.cs

```diff
@@ -0,0 +1,144 @@
+            byte[] input = new byte[algorithm.SecretKeySizeInBytes];
+
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.GenerateKey(algorithm));
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>.Empty, ReadOnlySpan<byte>.Empty));
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>.Empty, ReadOnlySpan<byte>.Empty));
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.ImportFromEncryptedPem(ReadOnlySpan<char>.Empty, ReadOnlySpan<byte>.Empty));
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.ImportFromEncryptedPem(ReadOnlySpan<char>.Empty, ReadOnlySpan<char>.Empty));
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.ImportFromPem(ReadOnlySpan<char>.Empty));
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.ImportPkcs8PrivateKey(ReadOnlySpan<byte>.Empty));
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, input.AsSpan(0, algorithm.PublicKeySizeInBytes)));
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.ImportSlhDsaSecretKey(algorithm, input.AsSpan(0, algorithm.SecretKeySizeInBytes)));
+            Assert.Throws<PlatformNotSupportedException>(() => SlhDsa.ImportSubjectPublicKeyInfo(ReadOnlySpan<byte>.Empty));
+        }
+
+        [Theory]
+        [MemberData(nameof(SlhDsaTestData.AlgorithmsData), MemberType = typeof(SlhDsaTestData))]
+        public static void AlgorithmMatches_GenerateKey(SlhDsaAlgorithm algorithm)
+        {
+            AssertThrowIfNotSupported(() =>
+            {
+                using SlhDsa slhDsa = SlhDsa.GenerateKey(algorithm);
+                Assert.Equal(algorithm, slhDsa.Algorithm);
+            });
+
+            AssertImportPublicKey(import =>
+                AssertThrowIfNotSupported(() =>
+                    Assert.Equal(algorithm, import().Algorithm)), algorithm, new byte[algorithm.PublicKeySizeInBytes]);
+
+            AssertImportSecretKey(import =>
+                AssertThrowIfNotSupported(() =>
+                    Assert.Equal(algorithm, import().Algorithm)), algorithm, new byte[algorithm.SecretKeySizeInBytes]);
+        }
+
+        private static void AssertImportPublicKey(Action<Func<SlhDsa>> test, SlhDsaAlgorithm algorithm, ReadOnlyMemory<byte> publicKey) =>
+            AssertImportPublicKey(test, test, algorithm, publicKey);
+
+        private static void AssertImportPublicKey(Action<Func<SlhDsa>> testDirectCall, Action<Func<SlhDsa>> testEmbeddedCall, SlhDsaAlgorithm algorithm, ReadOnlyMemory<byte> publicKey)
+        {
+            if (publicKey.Length == 0)
+            {
+                testDirectCall(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, []));
+                testDirectCall(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, ReadOnlySpan<byte>.Empty));
```

Comment:
This was my fault from earlier, but these two lines are the same.  Yay magic syntax.

The first one should be `Array.Empty<byte>()` so we are testing both (null, 0) and (non-null, 0) through the P/Invokes.